3rdParty/iresearch/core/store/store_utils.hpp

```diff
@@ -486,12 +452,55 @@ class IRESEARCH_API bytes_ref_input : public index_input {
     return irs::vread<uint32_t>(pos_);
   }
 
-  virtual int64_t checksum(size_t offset) const override final;
+  virtual int64_t checksum(size_t offset) const override;
 
  private:
   bytes_ref data_;
-  const byte_type* pos_{ data_.begin() };
-}; // bytes_ref_input
+  const byte_type* pos_{data_.begin()};
+};  // bytes_ref_input
+
+// same as bytes_ref_input but with support of adress remapping
+// usable when original data offses needs to be persistent
+// NOTE: remapped data blocks may have gaps but should not overlap!
+class IRESEARCH_API remapped_bytes_ref_input : public bytes_ref_input {
+ public:
+  using mapping_value = std::pair<size_t, size_t>;
+  using mapping = std::vector<mapping_value>;
+
+  explicit remapped_bytes_ref_input(const bytes_ref& data, mapping&& mapping)
```

Comment:
Maybe store mapping outside so we don't need it for every stream?

Reply:
That would complicate ownership (we will need shared pointers or so). And we will move out of remapped_bytes_ref_input mapping sort, or somehow ensure it i not resorted. So I suggest for now to not do this